onecodex/viz/_distance.py

```diff
@@ -42,19 +42,33 @@ def _compute_distance(self, rank, metric):
         return distances
 
     def _cluster_by_sample(
-        self, rank=Rank.Auto, metric=BetaDiversityMetric.BrayCurtis, linkage=Linkage.Average
+        self,
+        all_nan_classification_ids=None,
```

Comment:
I thin you can get this from `self.all_nan_classification_ids` since that's defined on `AnalysisMixin` which works for both `ClassificationsDataFrame` and `SampleCollection`

Reply:
True, but this is used by `plot_distance`, in which we actually want to exclude the samples/classifications entirely from the plot (in a separate PR). Since the behavior is different, and because we actually use more than just the `ids_in_order`, I wanted to keep this as a parameter so that `plot_distance` could _not_ pass anything and keep its current behavior for now. Let me know if you have thoughts though or another idea about handling these two cases.